### PR TITLE
Tons of design polish

### LINKED ISSUE #110: Dedicated "clear filter" button
It would be mega-helpful to have some sort of "clear filter" button that doesn't require 3 taps to get to:
hamburger icon -> gear icon -> clear filters

Maybe this button ONLY shows up when you have filters applied, which in turn solves an ancillary problem: knowing you have a filter applied when viewing your items.

Currently there is not indication of whether or not a filter is applied other than by drilling down on every filter menu.

### LINKED ISSUE #148: Improve Sign In Prompt
I posted this on Reddit but thought it would be more appropriate here:

You could clarify the text more. Make it more of a welcome screen instead displaying it as an error. 

Something like:  "The app has been updated. In order to continue,  please sign-in with your PSN/MS credentials" 

So text that explains why stuff is happening, appearing or why some actions are needed to be done is a good way to avoid confusion. Make sure to clarify that the sign-in thing is "by design". The word "error" does really suggest it's some kind of bug/not meant to happen. 

"Please wait, your inventory is loading" is a great message you added in 2.3.8! It really does give you more feedback on what is happening.

### LINKED ISSUE #176: Weapon detail not working properly on iPhone
image

I get this when I long press a weapon in the iPhone 6 iOS 8.3.

### PR DESCRIPTION
**Pull request by bluetidepro**
You'll prob want to pull this down and play with it locally to see all the design changes. There is a ton.

Feel free to leave feedback. This also fixes a few issues:

- Close #148 
- Close #176
- Close #110
- Some of #169 is touched
- Some of #66 is touched

Also - FYI, if you look at this PR this way: https://github.com/dasilva333/TowerGhostForDestiny/pull/183/files?w=1 - it will get rid of the white space changes.

### CODE REVIEW

**Comment by bluetidepro** [comment]
@dasilva333 I think this **MAY** need to update the version stuff? I'll let you handle that after the PR is merged :) haha

File: www/js/app.js [hunk 1]
```diff
@@ -1278,11 +1278,19 @@ var app = new(function() {
...
-        $(window).scroll(_.throttle(self.quickIconHighlighter, 500));
+        $(window).resize(_.throttle(self.bucketSizeHandler, 400));
+        $(window).resize(_.throttle(self.quickIconHighlighter, 400));
+        $(window).scroll(_.throttle(self.quickIconHighlighter, 400));
         self.whatsNew();
         ko.applyBindings(self);
+        
+        // Refresh Spinner
+        $('.spinner .btn:first-of-type').on('click', function() {
+            app.refreshSecond = app.refreshSeconds(parseInt(app.refreshSeconds()) + 1);
```
Comment (dasilva333) [inline www/js/app.js]: Still wrote it wrong, I have to go in and fix this after

Comment (bluetidepro) [inline www/js/app.js]: @dasilva333 What do you mean? That was working properly for me?

**Comment by bluetidepro** [comment]
@dasilva333 I separated out a few more commits to the PR for some of the other stuff we talked about. I thought it would be a better idea to not have those project type changes done in the same commit as my design stuff. Let me know if you have any final questions/comments/fixes before you merge! :)

If it's all good to go.... YAY! :fireworks:

**Pull request merged**